### ISSUE TITLE
parse conditonal translation `@if` statements

### ISSUE THREAD

**Issue by mighdoll** [wgsl-tooling-wg/wesl-js#2]
see [conditional translation spec](https://github.com/wgsl-tooling-wg/wesl-spec/blob/main/ConditionalTranslation.md)

**Comment by mighdoll** [comment]
Current plan:
- [ ] Grammar allows `@if` statements throughout
- [ ] Grammar parses if-expressions (expressions inside of `@if` statements)
- [ ] eval() for if-expressions to determines validity with current runtime conditions
- [ ] Collection tracks active if-expressions including parent elements
- [ ] Collection records active if-expressions with AbstractElems and Idents
- [ ] LowerAndEmit omits inactive expressions
- [ ] BindIdents skips inactive idents
- [ ] Celebrate!

**Comment by mighdoll** [comment]
revised title and task list to split out #105

**Comment by k2d222** [comment]
right, the spec doesn't clearly cover this case. I'm not sure if my impl supports it. I think we can keep it out of spec for now. Or explicitly disallow it.

**Comment by mighdoll** [comment]
tests pass, all known cases handled!
